lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -130,10 +131,11 @@ public ParquetReader(
             ParquetDataSource dataSource,
             DateTimeZone timeZone,
             AggregatedMemoryContext memoryContext,
-            ParquetReaderOptions options)
+            ParquetReaderOptions options,
+            Predicate parquetPredicate)
             throws IOException
     {
-        this(fileCreatedBy, fields, blocks, firstRowsOfBlocks, dataSource, timeZone, memoryContext, options, null, null);
+        this(fileCreatedBy, fields, blocks, firstRowsOfBlocks, dataSource, timeZone, memoryContext, options, parquetPredicate, nCopies(blocks.size(), Optional.empty()));
```

Comment:
Let's add `requireNonNull` to parquetPredicate and columnIndexStore now that they can't be null.
It would be better make parquetPredicate an optional and send Optional.empty from iceberg to avoid executing `this.filter = parquetPredicate.toParquetFilter(timeZone)`